frontend/src/app/Routes.tsx

```diff
@@ -31,14 +34,18 @@ const Routes: React.FC = () => {
         <Route path="/" exact component={InstalledApplications} />
         <Route path="/explore" exact component={ExploreApplications} />
         <Route path="/resources" exact component={LearningCenterPage} />
-        <Route path="/notebookController">
-          <NotebookControllerRoutes />
-        </Route>
-        <Route
-          path="/notebook/:namespace/:notebookName/logout"
-          exact
-          component={NotebookLogoutRedirectPage}
-        />
+        {isAllowed && (
+          <Route path="/notebookController">
+            <NotebookControllerRoutes />
+          </Route>
+        )}
+        {isAllowed && (
+          <Route
+            path="/notebook/:namespace/:notebookName/logout"
+            exact
+            component={NotebookLogoutRedirectPage}
+          />
+        )}
```

Comment:
Hmmm, I wonder how we should approach this. Is `isAllowed` an official stance on "allowing access to do stuff"?

So we have the three states:
- Anyone who can reach the Dashboard -- eg this is the Application Enabled/Installed & the Resources
- Anyone who has permissions to create things like projects (coming) / notebooks / etc -- eg this is the KFNBC
- Anyone who has administrative permissions -- eg any other item that only admins can do

If I nailed it -- I think we should probably break out the routes into their own files/sections and that will setup foundation for us to put new pages in one of the 3 spots.

Reply:
I will wait with asking for re-review until this comment (and the comment under it) is checked by @jedemo. I agree but I want to be sure what direction to take here, same with the naming.

Reply:
Let's delay this from this PR... I've created: #408